linera-core/src/client/mod.rs

```diff
@@ -2469,60 +2356,131 @@ where
             }
         }
         self.update_from_info(&info);
```

Comment:
Why do we have to update explicitly BTW? we've made the query for `chain_info_with_manager_values` earlier, can't we update `self` there immediately?

Reply:
Since we may have updated the `info` inside the `if` branch above, there's no point: The update would already be outdated.

Reply:
Can we do the update inside `chain_info_with_manager_values`? Or add a wrapper that does the two things? I wonder: isn't it (in general) correct/safe thing to do? every time we fetch the latest `chain_info*` we should update our local state/view (`self.update_from_info`) based on that? Even if it's not technically necessary by doing that we can "forget" about when to call it when it _Is_ necessary.